public/store/FindingsStore.ts

```diff
@@ -111,25 +116,60 @@ export class FindingsStore implements IFindingsStore {
     return [];
   };
 
-  public getFindingsPerDetector = async (detectorId: string): Promise<Finding[]> => {
-    let allFindings: Finding[] = [];
+  public getFindingsPerDetector = async (
+    detectorId: string,
+    detector: DetectorHit,
+    abort: AbortSignal,
+    duration?: Duration,
+    onPartialFindingsFetched?: (findings: FindingItemType[]) => void
+  ): Promise<FindingItemType[]> => {
+    let allFindings: FindingItemType[] = [];
     const findingsSize = 10000;
-    const firstGetFindingsRes = await this.service.getFindings({
+    const getFindingsQueryParams = {
       detector_id: detectorId,
       startIndex: 0,
       size: findingsSize,
-    });
+      startTime: duration?.startTime,
+      endTime: duration?.endTime
+    }
+
+    if (abort.signal) {
+      return allFindings;
+    }
+
+    const firstGetFindingsRes = await this.service.getFindings(getFindingsQueryParams);
 
     if (firstGetFindingsRes.ok) {
-      allFindings = [...firstGetFindingsRes.response.findings];
+      const extendedFindings = this.extendFindings(firstGetFindingsRes.response.findings, detector);
+      onPartialFindingsFetched?.(extendedFindings);
+      allFindings = [...extendedFindings];
       let remainingFindings = firstGetFindingsRes.response.total_findings - findingsSize;
       let startIndex = findingsSize + 1;
       const getFindingsPromises: Promise<ServerResponse<GetFindingsResponse>>[] = [];
 
       while (remainingFindings > 0) {
```

Comment:
`while (remainingFindings > 0 && !abort.signal) `, then i think then we can remove all the `if (abort.signal)` snippets and we can use Promise.all to return  all the Findings using `await Promise.all(getFindingsPromises);`

Reply:
Since I check for the signal multiple times inside, I don't want to add it to just the while check. Also, once aborted, we don't want to do any more processing so doesn't make sense to see if the promises settled. We simply return.